### DISCUSSION TITLE
Dynamic app layout with wrapped component in HOC

### DISCUSSION THREAD

**Discussion by overide** [vercel/next.js#11337] [Help] [answered]
Hi all,

I'm trying to get **[dynamic app layout](https://github.com/zeit/next.js/tree/canary/examples/with-dynamic-app-layout)** example working with my page wrapped in HOC. My page export is like this -`export default connect()(withAuth(Profile));`

In this case, I'm getting **WrappedComponent** instead of **Profile** which makes sense but how can I access the **Layout** property from the Profile component in _app.js? I'm looking for a generic solution which can work with any number of HOCs

Thanks!

**Comment by overide** [comment]
Hi @timneutkens I'll really appreciate it if you can provide some guidance, I'm kinda stuck at this point. Thanks

**Comment by aaronjensen** [comment, marked as answer]
It's the responsibility of HOCs to hoist statics of the wrapped component. If they do not, they are not proper HOCs and this is not expected to work. 

See: https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over

**Reply by overide** [reply]
Thanks @aaronjensen , this is indeed the right way, but my requirement was little bit more complex so I made the layout configurable through the page props.
